File: src/Array/FindPivotIndex/FindPivotIndex.java
class Solution {
    public int pivotIndex(int[] nums) {
        //1. Get Sum of the array
        int sum = 0;
        for(int num : nums){
            sum += num;
        }
        //2. Define the leftSum & rightSum
        int leftSum = 0, rightSum = sum;
        
        //Find the pivot index
        for(int i = 0; i < nums.length; i++){
            rightSum -= nums[i];
            if(leftSum == rightSum) return i;
            leftSum += nums[i];
        }
        //3. return -1 if can't find it
        return -1;
    }
}

File: src/Array/FindPivotIndex/FindPivotIndex.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        left_sum = [0] * n
        
        for i in range(1, n):
            left_sum[i] = left_sum[i - 1] + nums[i - 1]
        
        right_sum = [0] * n
        
        for i in reversed(range(n - 1)):
            right_sum[i] = right_sum[i + 1] + nums[i + 1]
            
        for i in range(n):
            if left_sum[i] == right_sum[i]: 
                return i
        
        return -1
